core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingInterceptor.java

```diff
@@ -129,20 +132,20 @@ private boolean performPartitionCheck(InvocationContext ctx, LocalFlagAffectedCo
    }
 
    private CompletableFuture<Void> handleDataReadCommand(InvocationContext ctx, DataCommand command) {
-      Object key = command.getKey();
       return ctx.onReturn((rCtx, rCommand, rv, throwable) -> {
+         DataCommand dataCommand = (DataCommand) rCommand;
          if (throwable != null) {
-            if (throwable instanceof RpcException && performPartitionCheck(rCtx, ((DataCommand) rCommand))) {
+            if (throwable instanceof RpcException && performPartitionCheck(rCtx, dataCommand)) {
                // We must have received an AvailabilityException from one of the owners.
                // There is no way to verify the cause here, but there isn't any other way to get an invalid
                // get response.
-               throw log.degradedModeKeyUnavailable(key);
+               throw log.degradedModeKeyUnavailable(dataCommand.getKey());
```

Comment:
Sorry, I wrote this thinking about ISPN-6795 and I without checking `ClusteredGetValidityFilter` again. Looks good, maybe just squash the commits a bit?